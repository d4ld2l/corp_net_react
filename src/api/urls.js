// @flow

export const API_URL = '/api'
export const APP_SETTINGS = `${API_URL}/settings`
export const BIRTHDATES_NEAREST_URL = `${API_URL}/birth_dates/nearest`
export const CANDIDATES_URL = `${API_URL}/candidates`
export const CANDIDATE_URL = `${CANDIDATES_URL}/{{ id }}`
export const CANDIDATE_VACANCIES = `${API_URL}/candidate_vacancies/{{ id }}/comments`
export const ACCOUNTS_URL = `${API_URL}/accounts`
export const GET_TICKET_URL = `${ACCOUNTS_URL}/get_ticket`
export const PROFILE_URL = `${ACCOUNTS_URL}/{{ id }}`
export const PROFILE_PHOTOS_URL = `${ACCOUNTS_URL}/{{ accountId }}/account_photos`
export const PROFILE_PHOTO_URL = `${ACCOUNTS_URL}/{{ accountId }}/account_photos/ {{ id }}`
export const CONFIRM_SKILL_URL = `${API_URL}/account_skills/{{ id }}/confirm`
export const UNCONFIRM_SKILL_URL = `${API_URL}/account_skills/{{ id }}/unconfirm`
export const EVENTS_URL = `${API_URL}/events`
export const DICTIONARIS_EVENT_TYPE_URL = `${API_URL}/dictionaries/event_type`
export const DICTIONARIES_ACCOUNTS_URL = `${ACCOUNTS_URL}/dictionary`
export const DICTIONARIES_PROJECTS_URL = `${API_URL}/projects/dictionary`
export const DICTIONARIES_VACANCIES_URL = `${API_URL}/vacancies/dictionary`
export const DICTIONARIES_VACANCY_STAGE_GROUP_URL = `${API_URL}/dictionaries/vacancy_stage_group`
export const DICTIONARIES_NEWS_CATEGORY_URL = `${API_URL}/dictionaries/news_category`

export const DICTIONARIES_DEPARTMENTS_URL = `${API_URL}/dictionaries/department`
export const DICTIONARIES_DEPARTMENTS_NAMES_URL = `${API_URL}/dictionaries/department_names`
export const DICTIONARIES_PROFESSIONAL_SPECIALIZATIONS_URL = `${API_URL}/dictionaries/professional_specialization`
export const DICTIONARIES_CANDIDATE_SKILLS_URL = `${API_URL}/dictionaries/candidate_skill`

export const DICTIONARIES_TAG_URL = `${API_URL}/dictionaries/tag`
export const DICTIONARIES_RESUME_SOURCES_URL = `${API_URL}/dictionaries/resume_source`
export const DICTIONARIES_LANGUAGES_URL = `${API_URL}/dictionaries/language`
export const DICTIONARIES_LANGUAGES_LEVEL_URL = `${API_URL}/dictionaries/language_level`
export const DICTIONARIES_EDUCATION_LEVEL_URL = `${API_URL}/dictionaries/education_level`
export const DICTIONARIES_SPECIALIZATION_URL = `${API_URL}/dictionaries/professional_area`
export const DICTIONARIES_OFFICES_URL = `${API_URL}/dictionaries/office`
export const DICTIONARIES_POSITIONS_URL = `${API_URL}/dictionaries/position`
export const DICTIONARIES_SKILLS_URL = `${API_URL}/dictionaries/skill`
export const DICTIONARIES_CITIES_URL = `${API_URL}/dictionaries/accounts_city`
export const DICTIONARIES_STRUCTURE_UNITS_URL = `${API_URL}/dictionaries/structure_unit`
export const DICTIONARIES_CONTACT_TYPES_URL = `${API_URL}/dictionaries/contract_type`
export const DICTIONARIES_EMPLOYEE_STATES_URL = `${API_URL}/dictionaries/employee_state`
export const EVENT_URL = `${EVENTS_URL}/{{ id }}`
export const EVENT_PARTICIPANTS = `${API_URL}/event_participants`
export const DICTIONARIES_SEARCH_SKILLS_URL = `${API_URL}/skills`
export const TEMPLATE_STAGES_URL = `${API_URL}/template_stages`
export const NEWS_URL = `${API_URL}/news`
export const NEW_URL = `${NEWS_URL}/{{ id }}`
export const PROFILE_ME_URL = `${ACCOUNTS_URL}/me`
export const PERMISSIONS_MY_URL = `${API_URL}/permissions/my`
export const SURVEYS_URL = `${API_URL}/surveys`
export const SURVEY_URL = `${SURVEYS_URL}/{{ id }}`
export const SURVEY_RESULTS_URL = `${API_URL}/survey_results`
export const VACANCIES_URL = `${API_URL}/vacancies`
export const VACANCY_URL = `${API_URL}/vacancies/{{ id }}`
export const VACANCY_STAGES_URL = `${API_URL}/vacancies/{{ id }}/vacancy_stages`
export const VACANCY_CANDIDATES_URL = `${API_URL}/vacancies/{{ id }}/candidates`
export const VACANCY_CANDIDATE_URL = `${API_URL}/vacancies/{{ vacancyId }}/candidates/{{ candidateId }}`
export const TRANSFER_CANDIDATE_URL = `${API_URL}/vacancies/{{ vacancyId }}/candidates/transfer`
export const EMAIL_NOTIFICATION = `${API_URL}/send_resume_letters`
export const VACANCY_CHANGE_STATE_URL = `${VACANCY_URL}/change_state/{{ state }}`
export const DEPARTMENTS_TREE_URL = `${API_URL}/departments/tree`
export const DEPARTMENTS_URL = `${API_URL}/departments`
export const SERVICES_URL = `${API_URL}/services`
export const SERVICE_URL = `${API_URL}/services/{{ id }}`
export const BIDS_URL = `${API_URL}/bids`
export const BIDS_AUTHOR_URL = `${BIDS_URL}/author`
export const BIDS_EXECUTOR_URL = `${BIDS_URL}/executor`
export const BID_URL = `${BIDS_URL}/{{ id }}`
export const BID_STATES_URL = `${BIDS_URL}/{{ id }}/states`
export const BID_ALLOWED_STATES_URL = `${BIDS_URL}/{{ id }}/allowed_states`
export const BID_CHANGE_STATE_URL = `${BIDS_URL}/{{ id }}/change_state/{{ state }}`
export const BID_STB_PARTICIPANTS_STATE_URL = `${BIDS_URL}/{{ id }}/team_building_information_accounts`
export const CUSTOMERS_URL = `${API_URL}/customers`
export const CUSTOMERS_CONTACTS_URL = `${API_URL}/customers/{{ customerId }}/contacts`
export const CUSTOMERS_CONTACT_URL = `${API_URL}/customers/{{ customerId }}/contacts/{{ id }}`
export const COMMENT_CUSTOMERS_CONTACT_URL = `${API_URL}/customers/{{ customerId }}/contacts/{{ id }}/add_comment`
export const LEGAL_UNITS_URL = `${API_URL}/legal_units`
export const LEGAL_UNITS_DICTIONARY_URL = `${API_URL}/legal_units/dictionary`
export const PROJECTS_URL = `${API_URL}/projects`
export const PROJECTS_DICRTIONARY_URL = `${API_URL}/dictionaries/tag?entity=project`
export const PROJECT_URL = `${PROJECTS_URL}/{{ id }}`
export const PROFILES_PROJECT_URL = `${PROJECTS_URL}/{{ id }}/account_projects`
export const PROFILE_PROJECT_URL = `${PROJECTS_URL}/{{ id }}/account_projects/{{project_id}}`
export const FEEDS_URL = `${API_URL}/feed`
export const USERS_COUNTER_URL = `${API_URL}/visits/stats`
export const DISTRIBUTIONS_URL = `${API_URL}/mailing_lists`
export const DISTRIBUTION_URL = `${API_URL}/mailing_lists/{{ id }}`
export const COMPARISON_SAVE_BOTH = `${API_URL}/save_both`
export const COMPARISON_SAVE_ONE = `${API_URL}/save_one`
export const TASKS_URL = `${API_URL}/tasks`
export const TASK_URL = `${API_URL}/tasks/{{ id }}`
export const SUBTASKS_URL = `${API_URL}/tasks/{{ task_id }}/subtasks`
export const SUBTASK_URL = `${API_URL}/tasks/{{ task_id }}/subtasks/{{ id }}`
export const UPDATE_PASSWORD_URL = `${ACCOUNTS_URL}/update_password`
export const VACANCY_STAGES_GROUPS = `${API_URL}/vacancy_stage_groups/stats`
export const FILTERED_ANALYTICS_STAGES_GROUPS = `${API_URL}/vacancy_stage_groups/recruitment_vortex`
export const PROFILE_HR_URL = `${ACCOUNTS_URL}/{{ id }}?hr=true`
export const DISCUSSIONABLE_URL = `discussions`
export const DISCUSSIONS_URL = `${API_URL}/${DISCUSSIONABLE_URL}`
export const DISCUSSION_URL = `${API_URL}/${DISCUSSIONABLE_URL}/{{ id }}`
export const CANDIDATE_RATINGS_URL = `${API_URL}/candidate_ratings`
export const ASSESSMENT_SESSIONS_URL = `${API_URL}/assessment/sessions`
export const ASSESSMENT_SESSION_URL = `${API_URL}/assessment/sessions/{{id}}`
export const EVALUATIONS_ASSESSMENT_SESSION_URL = `${API_URL}/assessment/sessions/{{id}}/evaluations`
export const ASSESSMENT_SESSION_RESULT_URL = `${API_URL}/assessment/sessions/{{id}}/result`

export const AUTH_URL = '/auth'
export const RESET_NEW_PASSWORD_URL = `${AUTH_URL}/reset_password`
export const GENERATE_NEW_PASSWORD_URL = `${AUTH_URL}/generate_new_password_email`
export const SIGN_IN_URL = `${AUTH_URL}/signin`
export const LOGOUT_URL = `${AUTH_URL}/logout`

// ui
export const LOGGIN_SETTINGS_URL = `${AUTH_URL}/login_settings`
