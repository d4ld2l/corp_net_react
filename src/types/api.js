import type { AxiosPromise } from 'axios'

import type {
  BirthDatesNearestRaw,
  CandidateRaw,
  CandidateWithIncludesRaw,
  EmployeesRaw,
  EmployeeRaw,
  EventsRaw,
  EventRaw,
  EventTypesRaw,
  ParticipantsRaw,
  UserRaw,
  MilestonesGroupRaw,
  MilestoneTemplateRaw,
  NewRaw,
  NewsRaw,
  NewsCategoriesRaw,
  ProfileRaw,
  ProfileRoleRaw,
  RecruiterRoleInfoRaw,
  SurveysRaw,
  SurveyRaw,
  TagsRaw,
  VacancyRaw,
  VacanciesRaw,
  CandidateVacancyRaw,
  ResumeRaw,
  AdditionalContactRaw,
  ResumeWorkExperienceRaw,
  ResumeRecommendationRaw,
  LanguageSkillRaw,
  LanguageRaw,
  LanguageLevelRaw,
  ResumeEducationRaw,
  EducationLevelRaw,
  CandidatePermittedAttrsRaw,
  ResumeSourceRaw,
  ServicesRaw,
  ServiceRaw,
  BidsRaw,
  BidRaw,
} from './raws'

import type { DepartmentsRaw } from '../presenters/departments'

export type Api = {
  birthdates: {
    nearest: () => AxiosPromise<BirthDatesNearestRaw>,
  },
  candidates: {
    all: () => AxiosPromise<Array<CandidateWithIncludesRaw>>,
    select: (params: string) => AxiosPromise<Array<CandidateWithIncludesRaw>>,
    search: (params: CandidatePermittedAttrsRaw) => AxiosPromise<Array<CandidateWithIncludesRaw>>,
    advancedSearch: (
      params: CandidatePermittedAttrsRaw
    ) => AxiosPromise<Array<CandidateWithIncludesRaw>>,
    update: (
      id: number,
      params: CandidatePermittedAttrsRaw
    ) => AxiosPromise<CandidateWithIncludesRaw | { success: false }>,
    findById: (id: number) => AxiosPromise<CandidateWithIncludesRaw>,
    whereGroupId: (id: number) => AxiosPromise<Array<CandidateWithIncludesRaw>>,
    create: (
      params: CandidatePermittedAttrsRaw
    ) => AxiosPromise<CandidateWithIncludesRaw | { success: false }>,
    parsing: (params: {
      source: string,
      file: string,
    }) => AxiosPromise<CandidateWithIncludesRaw | { success: false }>,
  },
  news: {
    create: (params: Object) => AxiosPromise<NewRaw>,
    all: () => AxiosPromise<NewsRaw>,
    where: (params: { params: * }) => AxiosPromise<NewsRaw>,
    findById: (id: number) => AxiosPromise<NewRaw>,
    categories: () => AxiosPromise<NewsCategoriesRaw>,
  },
  employees: {
    findById: (id: number) => AxiosPromise<EmployeeRaw>,
    search: (params: { params: * }) => AxiosPromise<EmployeesRaw>,
    all: () => AxiosPromise<EmployeesRaw>,
  },
  events: {
    all: () => AxiosPromise<EventsRaw>,
    findById: (id: number) => AxiosPromise<EventRaw>,
    types: () => AxiosPromise<EventTypesRaw>,
    create: (params: Object) => AxiosPromise<EventRaw>,
    update: (id: number, params: Object) => AxiosPromise<EventRaw>,
    delete: (id: number) => AxiosPromise<{}>,
    searchParticipants: (q: string) => AxiosPromise<ParticipantsRaw>,
  },
  auth: {
    signin: (params: Object) => AxiosPromise<UserRaw>,
    logout: () => AxiosPromise<UserRaw>,
  },
  profiles: {
    me: () => AxiosPromise<ProfileRaw>,
  },
  permissions: {
    my: () => AxiosPromise<ProfileRoleRaw>,
  },
  surveys: {
    all: () => AxiosPromise<SurveysRaw>,
    findById: (id: number) => AxiosPromise<SurveyRaw>,
    search: (q: string) => AxiosPromise<SurveysRaw>,
    save: (params: Object) => AxiosPromise<SurveyRaw>,
  },
  users_admin: {
    where: (params: { params: * }) => AxiosPromise<RecruiterRoleInfoRaw>,
  },
  tags: {
    all: () => AxiosPromise<TagsRaw>,
  },
  vacancies: {
    all: () => AxiosPromise<VacanciesRaw>,
    myVacancies: () => AxiosPromise<VacanciesRaw>,
    withStages: (stage: string) => AxiosPromise<VacanciesRaw>,
    changeState: (id: number, state: string) => AxiosPromise<VacancyRaw>,
    findBy: (params: { vacancy: * }) => AxiosPromise<VacancyRaw>,
    update: (id: number, params: { vacancy: * }) => AxiosPromise<VacancyRaw>,
    findById: (id: number) => AxiosPromise<VacancyRaw>,
  },
  vacancy_stage_group: () => AxiosPromise<MilestonesGroupRaw>,
  template_stages: () => AxiosPromise<MilestoneTemplateRaw>,
  resume_sources: () => AxiosPromise<Array<ResumeSourceRaw>>,
  languages: () => AxiosPromise<Array<LanguageRaw>>,
  languagesLevel: () => AxiosPromise<Array<LanguageLevelRaw>>,
  educationLevel: () => AxiosPromise<Array<EducationLevelRaw>>,
  departments: {
    tree: () => AxiosPromise<DepartmentsRaw>,
  },
  services: {
    all: () => AxiosPromise<ServicesRaw>,
    find: () => AxiosPromise<ServiceRaw>,
  },
  bids: {
    all: () => AxiosPromise<BidsRaw>,
    find: () => AxiosPromise<BidRaw>,
  },
}
