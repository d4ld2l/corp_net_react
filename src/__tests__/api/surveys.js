import api from '../../api'

describe('API Surveys', () => {
  it('should fetch all surveys', async () => {
    const response = await api.surveys.all()
    expect(response.data).toMatchSnapshot()
  })

  it('should find survey by id', async () => {
    const response = await api.surveys.findById(9)
    expect(response.data).toMatchSnapshot()
  })
})
