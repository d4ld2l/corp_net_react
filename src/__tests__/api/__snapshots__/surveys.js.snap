// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`API Surveys should fetch all surveys 1`] = `
Array [
  Object {
    "background": "#38606e",
    "created_or_updated_at": "2017-09-27T20:42:43.324Z",
    "document": Object {
      "url": null,
    },
    "ends_at": "2018-04-13",
    "id": 9,
    "name": "Опрос рабочий (важный!)",
    "note": "Тут явно не хватает описания!
Хочется заранее поблагодарить всех, кто поучаствует в опросе!
Особенно - себя.",
    "participant": 5,
    "passed": true,
    "published_at": "2017-09-27T20:38:20.166Z",
    "publisher": Object {
      "email": "admin@example.com",
      "full_name": "Старший Системный Администратор",
      "id": 1,
      "roles": Array [
        Object {
          "created_at": "2017-07-19T12:13:10.681Z",
          "id": 1,
          "name": "admin",
          "updated_at": "2017-07-19T12:13:10.681Z",
        },
        Object {
          "created_at": "2017-07-25T19:11:25.045Z",
          "id": 6,
          "name": "user",
          "updated_at": "2017-07-25T19:11:25.045Z",
        },
        Object {
          "created_at": "2017-07-19T12:13:10.685Z",
          "id": 2,
          "name": "recruitment_recruiter",
          "updated_at": "2017-10-10T12:32:56.949Z",
        },
      ],
    },
    "questions_count": 3,
    "state": "Опубликован",
    "survey_type": "Простой",
    "symbol": Object {
      "medium": Object {
        "url": "/uploads/survey/symbol/9/medium_92d56434-f095-4b7e-9651-ab2b5878eece.jpg",
      },
      "small_medium": Object {
        "url": "/uploads/survey/symbol/9/small_medium_92d56434-f095-4b7e-9651-ab2b5878eece.jpg",
      },
      "thumb": Object {
        "url": "/uploads/survey/symbol/9/thumb_92d56434-f095-4b7e-9651-ab2b5878eece.jpg",
      },
      "url": "/uploads/survey/symbol/9/92d56434-f095-4b7e-9651-ab2b5878eece.jpg",
    },
  },
  Object {
    "background": "#ffffff",
    "created_or_updated_at": "2017-09-22T16:31:57.972Z",
    "document": Object {
      "url": null,
    },
    "ends_at": "2017-10-07",
    "id": 8,
    "name": "Очень сложный опрос",
    "note": "Очень сложный опрос",
    "participant": 4,
    "passed": true,
    "published_at": "2017-09-22T16:31:57.970Z",
    "publisher": Object {
      "email": "dmishukov@fil-it.ru",
      "full_name": "Мишуков Дмитрий Михайлович",
      "id": 6,
      "roles": Array [
        Object {
          "created_at": "2017-07-19T12:13:10.681Z",
          "id": 1,
          "name": "admin",
          "updated_at": "2017-07-19T12:13:10.681Z",
        },
        Object {
          "created_at": "2017-07-25T19:11:25.045Z",
          "id": 6,
          "name": "user",
          "updated_at": "2017-07-25T19:11:25.045Z",
        },
      ],
    },
    "questions_count": 1,
    "state": "Опубликован",
    "survey_type": "Сложный",
    "symbol": Object {
      "medium": Object {
        "url": "/uploads/survey/symbol/8/medium_daoffice.png",
      },
      "small_medium": Object {
        "url": "/uploads/survey/symbol/8/small_medium_daoffice.png",
      },
      "thumb": Object {
        "url": "/uploads/survey/symbol/8/thumb_daoffice.png",
      },
      "url": "/uploads/survey/symbol/8/daoffice.png",
    },
  },
  Object {
    "background": "#ffffff",
    "created_or_updated_at": "2017-10-11T12:36:28.343Z",
    "document": Object {
      "url": null,
    },
    "ends_at": "2017-10-12",
    "id": 7,
    "name": "Любимый фильм",
    "note": "<p>Любимый фильм</p>
",
    "participant": 0,
    "passed": false,
    "published_at": "2017-09-27T18:00:04.067Z",
    "publisher": Object {
      "email": "admin@example.com",
      "full_name": "Старший Системный Администратор",
      "id": 1,
      "roles": Array [
        Object {
          "created_at": "2017-07-19T12:13:10.681Z",
          "id": 1,
          "name": "admin",
          "updated_at": "2017-07-19T12:13:10.681Z",
        },
        Object {
          "created_at": "2017-07-25T19:11:25.045Z",
          "id": 6,
          "name": "user",
          "updated_at": "2017-07-25T19:11:25.045Z",
        },
        Object {
          "created_at": "2017-07-19T12:13:10.685Z",
          "id": 2,
          "name": "recruitment_recruiter",
          "updated_at": "2017-10-10T12:32:56.949Z",
        },
      ],
    },
    "questions_count": 1,
    "state": "Опубликован",
    "survey_type": "Простой",
    "symbol": Object {
      "medium": Object {
        "url": "/uploads/survey/symbol/7/medium_Jellyfish.jpg",
      },
      "small_medium": Object {
        "url": "/uploads/survey/symbol/7/small_medium_Jellyfish.jpg",
      },
      "thumb": Object {
        "url": "/uploads/survey/symbol/7/thumb_Jellyfish.jpg",
      },
      "url": "/uploads/survey/symbol/7/Jellyfish.jpg",
    },
  },
  Object {
    "background": "#9cba97",
    "created_or_updated_at": "2017-10-02T10:05:26.313Z",
    "document": Object {
      "url": null,
    },
    "ends_at": "2017-10-07",
    "id": 6,
    "name": "Определяем ваши вкусовые предпочтения",
    "note": "Древний опрос, определяющий ваши вкусовые предпочтения",
    "participant": 1,
    "passed": true,
    "published_at": "2017-09-27T18:38:17.251Z",
    "publisher": Object {
      "email": "admin@example.com",
      "full_name": "Старший Системный Администратор",
      "id": 1,
      "roles": Array [
        Object {
          "created_at": "2017-07-19T12:13:10.681Z",
          "id": 1,
          "name": "admin",
          "updated_at": "2017-07-19T12:13:10.681Z",
        },
        Object {
          "created_at": "2017-07-25T19:11:25.045Z",
          "id": 6,
          "name": "user",
          "updated_at": "2017-07-25T19:11:25.045Z",
        },
        Object {
          "created_at": "2017-07-19T12:13:10.685Z",
          "id": 2,
          "name": "recruitment_recruiter",
          "updated_at": "2017-10-10T12:32:56.949Z",
        },
      ],
    },
    "questions_count": 1,
    "state": "Опубликован",
    "survey_type": "Простой",
    "symbol": Object {
      "medium": Object {
        "url": "/uploads/survey/symbol/6/medium_Tulips.jpg",
      },
      "small_medium": Object {
        "url": "/uploads/survey/symbol/6/small_medium_Tulips.jpg",
      },
      "thumb": Object {
        "url": "/uploads/survey/symbol/6/thumb_Tulips.jpg",
      },
      "url": "/uploads/survey/symbol/6/Tulips.jpg",
    },
  },
  Object {
    "background": "#ffffff",
    "created_or_updated_at": "2017-09-21T10:20:00.907Z",
    "document": Object {
      "url": null,
    },
    "ends_at": "2017-09-21",
    "id": 4,
    "name": "Простой анонимный опрос без фона",
    "note": "2 вопроса с 1м вариантом ответа из списка и своим вариантом",
    "participant": 4,
    "passed": true,
    "published_at": "2017-09-21T10:20:00.906Z",
    "publisher": Object {
      "email": "admin@example.com",
      "full_name": "Старший Системный Администратор",
      "id": 1,
      "roles": Array [
        Object {
          "created_at": "2017-07-19T12:13:10.681Z",
          "id": 1,
          "name": "admin",
          "updated_at": "2017-07-19T12:13:10.681Z",
        },
        Object {
          "created_at": "2017-07-25T19:11:25.045Z",
          "id": 6,
          "name": "user",
          "updated_at": "2017-07-25T19:11:25.045Z",
        },
        Object {
          "created_at": "2017-07-19T12:13:10.685Z",
          "id": 2,
          "name": "recruitment_recruiter",
          "updated_at": "2017-10-10T12:32:56.949Z",
        },
      ],
    },
    "questions_count": 2,
    "state": "Опубликован",
    "survey_type": "Простой",
    "symbol": Object {
      "medium": Object {
        "url": "/uploads/survey/symbol/4/medium_%D0%B2%D1%8B%D0%BB%D0%B8%D1%82%D0%B0%D1%8F_%D1%8F_.gif",
      },
      "small_medium": Object {
        "url": "/uploads/survey/symbol/4/small_medium_%D0%B2%D1%8B%D0%BB%D0%B8%D1%82%D0%B0%D1%8F_%D1%8F_.gif",
      },
      "thumb": Object {
        "url": "/uploads/survey/symbol/4/thumb_%D0%B2%D1%8B%D0%BB%D0%B8%D1%82%D0%B0%D1%8F_%D1%8F_.gif",
      },
      "url": "/uploads/survey/symbol/4/%D0%B2%D1%8B%D0%BB%D0%B8%D1%82%D0%B0%D1%8F_%D1%8F_.gif",
    },
  },
  Object {
    "background": "#ffffff",
    "created_or_updated_at": "2017-09-11T14:19:22.257Z",
    "document": Object {
      "url": null,
    },
    "ends_at": "2017-09-27",
    "id": 3,
    "name": "Сложный анонимный опрос",
    "note": "какое-то ",
    "participant": 5,
    "passed": true,
    "published_at": "2017-09-11T14:19:22.255Z",
    "publisher": Object {
      "email": "admin@example.com",
      "full_name": "Старший Системный Администратор",
      "id": 1,
      "roles": Array [
        Object {
          "created_at": "2017-07-19T12:13:10.681Z",
          "id": 1,
          "name": "admin",
          "updated_at": "2017-07-19T12:13:10.681Z",
        },
        Object {
          "created_at": "2017-07-25T19:11:25.045Z",
          "id": 6,
          "name": "user",
          "updated_at": "2017-07-25T19:11:25.045Z",
        },
        Object {
          "created_at": "2017-07-19T12:13:10.685Z",
          "id": 2,
          "name": "recruitment_recruiter",
          "updated_at": "2017-10-10T12:32:56.949Z",
        },
      ],
    },
    "questions_count": 1,
    "state": "Опубликован",
    "survey_type": "Сложный",
    "symbol": Object {
      "medium": Object {
        "url": "/uploads/survey/symbol/3/medium_dobro-B6MZ37.jpg",
      },
      "small_medium": Object {
        "url": "/uploads/survey/symbol/3/small_medium_dobro-B6MZ37.jpg",
      },
      "thumb": Object {
        "url": "/uploads/survey/symbol/3/thumb_dobro-B6MZ37.jpg",
      },
      "url": "/uploads/survey/symbol/3/dobro-B6MZ37.jpg",
    },
  },
  Object {
    "background": "#bcd1c5",
    "created_or_updated_at": "2017-09-18T14:49:56.103Z",
    "document": Object {
      "url": null,
    },
    "ends_at": "2017-09-30",
    "id": 2,
    "name": "Простой опрос с 2 вопросами, фоном и эмблемой",
    "note": "2 вопроса внутри, 1 с картинкой и один - без. Свой вариант в случае без картинки",
    "participant": 3,
    "passed": true,
    "published_at": "2017-09-11T13:42:07.375Z",
    "publisher": Object {
      "email": "admin@example.com",
      "full_name": "Старший Системный Администратор",
      "id": 1,
      "roles": Array [
        Object {
          "created_at": "2017-07-19T12:13:10.681Z",
          "id": 1,
          "name": "admin",
          "updated_at": "2017-07-19T12:13:10.681Z",
        },
        Object {
          "created_at": "2017-07-25T19:11:25.045Z",
          "id": 6,
          "name": "user",
          "updated_at": "2017-07-25T19:11:25.045Z",
        },
        Object {
          "created_at": "2017-07-19T12:13:10.685Z",
          "id": 2,
          "name": "recruitment_recruiter",
          "updated_at": "2017-10-10T12:32:56.949Z",
        },
      ],
    },
    "questions_count": 2,
    "state": "Опубликован",
    "survey_type": "Простой",
    "symbol": Object {
      "medium": Object {
        "url": "/uploads/survey/symbol/2/medium_2013-12-29_16-25-00.JPG",
      },
      "small_medium": Object {
        "url": "/uploads/survey/symbol/2/small_medium_2013-12-29_16-25-00.JPG",
      },
      "thumb": Object {
        "url": "/uploads/survey/symbol/2/thumb_2013-12-29_16-25-00.JPG",
      },
      "url": "/uploads/survey/symbol/2/2013-12-29_16-25-00.JPG",
    },
  },
  Object {
    "background": "#ffffff",
    "created_or_updated_at": "2017-09-07T16:39:34.379Z",
    "document": Object {
      "url": null,
    },
    "ends_at": "2017-09-30",
    "id": 1,
    "name": "Удовлетворённость SHR",
    "note": "Опрос для проверки удовлетворённости SHR",
    "participant": 2,
    "passed": true,
    "published_at": "2017-09-07T16:39:34.377Z",
    "publisher": Object {
      "email": "dmishukov@fil-it.ru",
      "full_name": "Мишуков Дмитрий Михайлович",
      "id": 6,
      "roles": Array [
        Object {
          "created_at": "2017-07-19T12:13:10.681Z",
          "id": 1,
          "name": "admin",
          "updated_at": "2017-07-19T12:13:10.681Z",
        },
        Object {
          "created_at": "2017-07-25T19:11:25.045Z",
          "id": 6,
          "name": "user",
          "updated_at": "2017-07-25T19:11:25.045Z",
        },
      ],
    },
    "questions_count": 1,
    "state": "Опубликован",
    "survey_type": "Простой",
    "symbol": Object {
      "medium": Object {
        "url": "/uploads/survey/symbol/1/medium_abe_simpson.png",
      },
      "small_medium": Object {
        "url": "/uploads/survey/symbol/1/small_medium_abe_simpson.png",
      },
      "thumb": Object {
        "url": "/uploads/survey/symbol/1/thumb_abe_simpson.png",
      },
      "url": "/uploads/survey/symbol/1/abe_simpson.png",
    },
  },
]
`;

exports[`API Surveys should find survey by id 1`] = `
Object {
  "anonymous": false,
  "background": "#38606e",
  "document": Object {
    "url": null,
  },
  "ends_at": "2018-04-13",
  "id": 9,
  "name": "Опрос рабочий (важный!)",
  "note": "Тут явно не хватает описания!
Хочется заранее поблагодарить всех, кто поучаствует в опросе!
Особенно - себя.",
  "passed": true,
  "questions": Array [
    Object {
      "ban_own_answer": true,
      "id": 12,
      "image_small_medium_url": null,
      "offered_variants": Array [
        Object {
          "id": 23,
          "wording": "Темное",
        },
        Object {
          "id": 24,
          "wording": "Светлое",
        },
      ],
      "wording": "Темное или светлое?",
    },
    Object {
      "ban_own_answer": false,
      "id": 13,
      "image_small_medium_url": null,
      "offered_variants": Array [
        Object {
          "id": 25,
          "wording": "рыба",
        },
        Object {
          "id": 26,
          "wording": "сухари",
        },
      ],
      "wording": "Сухарики или рыба?",
    },
    Object {
      "ban_own_answer": false,
      "id": 14,
      "image_small_medium_url": null,
      "offered_variants": Array [
        Object {
          "id": 27,
          "wording": "Да, почему бы и нет?",
        },
        Object {
          "id": 28,
          "wording": "Я не из \\"этих\\"! Уберите это!",
        },
      ],
      "wording": "Может сидру?",
    },
  ],
  "questions_count": 3,
  "survey_type": "Простой",
  "symbol": Object {
    "medium": Object {
      "url": "/uploads/survey/symbol/9/medium_92d56434-f095-4b7e-9651-ab2b5878eece.jpg",
    },
    "small_medium": Object {
      "url": "/uploads/survey/symbol/9/small_medium_92d56434-f095-4b7e-9651-ab2b5878eece.jpg",
    },
    "thumb": Object {
      "url": "/uploads/survey/symbol/9/thumb_92d56434-f095-4b7e-9651-ab2b5878eece.jpg",
    },
    "url": "/uploads/survey/symbol/9/92d56434-f095-4b7e-9651-ab2b5878eece.jpg",
  },
}
`;
